### PR TITLE
[Improve][Standalone] Standalone Add param of --metadata-url for runing with metadata

### PR DESCRIPTION
**Pull request by liangyuanpeng**
### Motivation

Standalone support running with metadata, like etcd, And rocksdb still by default.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: Yes, Standalone support running with metadata store, but rocksdb metadataStore by default.

### Documentation

Need to update docs? 

- [] `doc-required` 
  
- [x] `doc-not-needed` 
  
- [ ] `doc` 

- [ ] `doc-complete`

### CODE REVIEW

**Comment by codelipenghui** [comment]
@liangyuanpeng Any reason to support using separate metadata for standalone? The standalone is designed for development, not for production.

**Review by nodece — APPROVED** [review]

**Comment by nodece** [comment]
I think @liangyuanpeng wants to use other metadata.

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandalone.java [hunk 1]
```diff
@@ -434,10 +438,13 @@ public void close() {
...
-    private void startBookieWithRocksDB() throws Exception {
-        log.info("Starting BK with RocksDb metadata store");
-        String metadataStoreUrl = "rocksdb://" + Paths.get(metadataDir).toAbsolutePath();
+    private void startBookieWithMetadataStore() throws Exception {
+        if(StringUtils.isBlank(metadataStoreUrl)){
+            log.info("Starting BK with RocksDb metadata store");
+            metadataStoreUrl = "rocksdb://" + Paths.get(metadataDir).toAbsolutePath();
+        }else{
+            log.info("Starting BK with metadata store:",metadataStoreUrl);
+        }
```
Comment (tisonkun) [inline pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandalone.java:447]: Please format this code block according to our code style settings. Briefly, add necessary blanks.

**Review by tisonkun — CHANGES_REQUESTED** [review]
Please update the description according to the placeholder.

**Comment by liangyuanpeng** [comment]
> 

Thanks for your reply, Yes, Standalone is more used to development and we can use other metadata to development for check the metadata after support running other metadata. Actually, i'm really do it, Base on etcd metadata store for write the dev code and not zookeeper.

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandalone.java (hunk 1 above)
Comment (liangyuanpeng) [reply pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandalone.java:447]: Thanks for your review and i'm busy for otherthing, pick it up in  the next week maybe.

**Comment by liangyuanpeng** [comment]
@codelipenghui  @tisonkun  PTAL.

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandalone.java [hunk 2]
```diff
@@ -52,6 +52,7 @@
 import org.apache.pulsar.packages.management.storage.filesystem.FileSystemPackagesStorageProvider;
 import org.apache.pulsar.zookeeper.LocalBookkeeperEnsemble;
 
+
```
Comment (tisonkun) [inline pulsar-broker/src/main/java/org/apache/pulsar/PulsarStandalone.java:55]: Revert

**Review by tisonkun — APPROVED** [review]
LGTM. Minor comments inline.

**Comment by tisonkun** [comment]
@liangyuanpeng Pulsar CI has been refactored a lot. You can merge the latest master to run CI tasks.

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by liangyuanpeng** [comment]
@codelipenghui  PTAL,Thanks.

**Comment by nodece** [comment]
Ping @codelipenghui, could you review this PR?

**Review by Jason918 — APPROVED** [review]
LGTM

**Pull request merged**